addons/account_fleet/models/account_move.py

```diff
@@ -34,6 +28,13 @@ class AccountMoveLine(models.Model):
     # used to decide whether the vehicle_id field is editable
     need_vehicle = fields.Boolean(compute='_compute_need_vehicle')
 
+    @api.onchange('vehicle_id')
+    def _onchange_vehicle(self):
```

Comment:
Onchanges are only called in the interface, this seems to implement business logic, so should be a compute method field in my opinion right ?